Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs

```diff
@@ -658,3 +697,12 @@ public enum AmmoVisuals : byte
     MagLoaded,
     BoltClosed,
 }
+
+[ByRefEvent]
+public record struct RecoilKickAttemptEvent() : IInventoryRelayEvent
+{
+    // TODO CENT Make this relayed
+    public float Effectiveness = 1.0f;
+
+    public SlotFlags TargetSlots => SlotFlags.WITHOUT_POCKET;
+}
```

Comment:
This would be used by magboots, noslips, etc. to modify the effect of the kick.
I should probably rename `Effectiveness` since it's meaningless, but magboots would set that to 0.0,  nullifying the kick.